protobuf-converter/src/main/java/io/confluent/connect/protobuf/ProtobufData.java

```diff
@@ -944,11 +954,19 @@ private String dataTypeFromConnectSchema(
         return useWrapperForNullables && schema.isOptional()
             ? PROTOBUF_INT32_WRAPPER_TYPE : FieldDescriptor.Type.INT32.toString().toLowerCase();
       case INT32:
+        defaultType = FieldDescriptor.Type.INT32.toString().toLowerCase();
+        if (schema.parameters() != null && schema.parameters().containsKey(PROTOBUF_TYPE_PROP)) {
+          defaultType = schema.parameters().get(PROTOBUF_TYPE_PROP);
+        }
         return useWrapperForNullables && schema.isOptional()
-            ? PROTOBUF_INT32_WRAPPER_TYPE : FieldDescriptor.Type.INT32.toString().toLowerCase();
+            ? PROTOBUF_INT32_WRAPPER_TYPE : defaultType;
       case INT64:
+        defaultType = FieldDescriptor.Type.INT64.toString().toLowerCase();
+        if (schema.parameters() != null && schema.parameters().containsKey(PROTOBUF_TYPE_PROP)) {
+          defaultType = schema.parameters().get(PROTOBUF_TYPE_PROP);
+        }
         return useWrapperForNullables && schema.isOptional()
-            ? PROTOBUF_INT64_WRAPPER_TYPE : FieldDescriptor.Type.INT64.toString().toLowerCase();
+            ? PROTOBUF_INT64_WRAPPER_TYPE : defaultType;
```

Comment:
Depending on defaultType, this can return one of PROTOBUF_INT64_WRAPPER_TYPE, PROTOBUF_UINT32_WRAPPER_TYPE, and PROTOBUF_UINT64_WRAPPER_TYPE